Add Plotter2D class for 2D visualization with comprehensive tests

jerrypaulvarghese opened the pull request:
## Changes
- Added `Plotter2D` class in `fol/tools/plotter.py` for flat 2D mesh visualization
- Added comprehensive unit tests in `tests/unit/test_plotter2d.py` (9 tests, all passing)

## Features
- Reuses Plotter3D configuration
- Supports optional in-plane warping
- Auto-selects best sample by minimum error
- Generates 3-panel overview (FOL |U|, REF |U|, |ΔU|)

## Testing
All tests passing

jerrypaulvarghese committed:
Plotter functions_2D_3D

jerrypaulvarghese committed:
Add files via upload

jerrypaulvarghese committed:
Add 3D plotting support with fol.plotter.Plotter3D and inference modules

jerrypaulvarghese committed:
added mixed sample training mechanical_box example  with Plotter3D generated results

jerrypaulvarghese committed:
Plotter3D tests have passed

jerrypaulvarghese committed:
Merge branch 'main' of https://github.com/RezaNajian/implicit_fol into 3D_Tetra-ifol

jerrypaulvarghese committed:
Delete plotter directory

jerrypaulvarghese committed:
Add pyvista to install_requires for Plotter3D

jerrypaulvarghese committed:
Update ci.yml

jerrypaulvarghese committed:
Update ci.yml

jerrypaulvarghese committed:
Update ci.yml

jerrypaulvarghese committed:
Update ci.yml

jerrypaulvarghese committed:
Update ci.yml

jerrypaulvarghese committed:
Update ci.yml

jerrypaulvarghese committed:
Update setup.py

jerrypaulvarghese committed:
Update plotter.py

jerrypaulvarghese committed:
Merge pull request #6 from RezaNajian/main

Update Branch

jerrypaulvarghese committed:
Update usefull_functions.py with Neww se of Dirchlet BCS

Supports torsion, uniform stretch / compression, simple shear.

jerrypaulvarghese committed:
git pushMerge remote-tracking branch 'origin/main' into 3D_Tetra-ifo
Merge origin/main into 3D_Tetra-ifol: migrate setup.py → pyproject.toml

jerrypaulvarghese committed:
update

jerrypaulvarghese committed:
Add pyvista to dependencies in pyproject.toml

jerrypaulvarghese committed:
Update pyproject.toml

jerrypaulvarghese committed:
Update CI.yml

jerrypaulvarghese committed:
Update test_plotter3d.py

jerrypaulvarghese committed:
Update test_plotter3d.py

jerrypaulvarghese committed:
Delete tested_samples directory

RezaNajian committed:
Update CI.yml

jerrypaulvarghese committed:
Update usefull_functions.py

jerrypaulvarghese committed:
Update usefull_functions.py with new Dirichlet BC helpers

jerrypaulvarghese committed:
Update plotter.py with 2D plotter


case_dir = os.path.join('.', "Folder_with_vtk_files")  
vtk_files = glob.glob(os.path.join(case_dir, "*.vtk"))
if vtk_files:
    vtk_path = vtk_files[0]
    config = {
        "cmap": "coolwarm",
        "u_fol_prefix": "U_FOL_",
        "u_fe_prefix": "U_HFE_",
        "output_image": "overview2d.png",
        "warp_factor_2d": 1.2,       #  
        "scalar_bar_args": {            #  
            "title": "",
            "vertical": True,
            "label_font_size": 20,
            "height": .9,
            "width": 0.09,
        },
        "zoom": 1.2,
        "clip": False,
        "show_edges": True,
        "window_size": (1600, 1000),
    }


    plotter = Plotter2D(vtk_path=vtk_path, config=config)
    plotter.render_all_panels()
else:
    print("No .vtk file found to visualize.")

jerrypaulvarghese committed:
Create Hyper_elastic_2d

jerrypaulvarghese committed:
Rename Hyper_elastic_2d.py to hyper_elasticity_2d.py

jerrypaulvarghese committed:
Compressible Neohookean

jerrypaulvarghese committed:
Merge branch 'main' into 3D_Tetra-ifol

RezaNajian reviewed (changes requested):
@jerrypaulvarghese Thanks for extending the plot function to 2D. I believe hyper_elasticity_2d.py was added by accident—please remove it. Also, could you add a test for the 2D plotter?

jerrypaulvarghese committed:
Merge remote-tracking branch 'origin/main' into 3D_Tetra-ifol

jerrypaulvarghese committed:
Add comprehensive unit tests for Plotter2D class

- Test 2D mesh validation (rejects non-flat meshes)
- Test initialization and configuration
- Test best sample selection by minimum error
- Test magnitude field creation from vector fields
- Test mesh warping functionality
- Test panel rendering (single and all panels)
- Test warp factor configuration
- Test error handling for missing required fields

All 9 tests passing successfully.

jerrypaulvarghese committed:
Remove hyper_elasticity_2d.py from PR - not ready for merge

jerrypaulvarghese commented:
> @jerrypaulvarghese Thanks for extending the plot function to 2D. I believe hyper_elasticity_2d.py was added by accident—please remove it. Also, could you add a test for the 2D plotter?

>@RezaNajian : Thank you very much for the review, I have revised the changes and I created the tests also, removed the hyper_elasticity_2d.py file mentioned. Kindly consider.

jerrypaulvarghese committed:
Remove solver convergence plotting function

Removed the plot_solver_convergence function and its documentation.
